server/src-lib/Hasura/Server/Query.hs

```diff
@@ -135,20 +135,19 @@ fetchLastUpdate = do
        ORDER BY occurred_at DESC LIMIT 1
           |] () True
   case l of
-    []           -> return Nothing
-    [(instId, occurredAt)] ->
-      return $ Just (InstanceId instId, occurredAt)
+    []  -> return Nothing
+    [i] -> return $ Just i
     -- never happens
-    _            -> throw500 "more than one row returned by query"
+    _   -> throw500 "more than one row returned by query"
 
 recordSchemaUpdate :: InstanceId -> Q.TxE QErr ()
 recordSchemaUpdate instanceId =
   liftTx $ Q.unitQE defaultTxErrorHandler [Q.sql|
-             INSERT INTO
-                  hdb_catalog.hdb_schema_update_event
-                  (instance_id, occurred_at)
-             VALUES ($1::uuid, DEFAULT)
-            |] (Identity $ getInstanceId instanceId) True
+             INSERT INTO hdb_catalog.hdb_schema_update_event
+               (instance_id, occurred_at) VALUES ($1::uuid, DEFAULT)
+             ON CONFLICT ((occurred_at IS NOT NULL))
+             DO UPDATE SET instance_id = $1::uuid, occurred_at = DEFAULT
```

Comment:
Using `INSERT` instead of `UPDATE` because after initialise the `hdb_catalog.hdb_schema_update_event` is empty and insert is required in that case. Using `ON CONFLICT` clause to `update` the already present single row.